algorithms/boyer_moore.go

```diff
@@ -0,0 +1,35 @@
+package algorithms
+
+func Search(needle []rune, haystack []rune) int {
+	n := len(needle)
+	if n == 0 {
+		return -1
+	}
```

Comment:
ich fände es persönlich schöner, wenn hier statt -1 ein error oder etwas in der art wie "needle must not be empty" zurückkäme

Reply:
Das ist ein ausgezeichneter Punkt... Ich erinnere mich daran, dass wir diesen Ansatz bereits zu implementieren begonnen, jedoch abgebrochen haben. Jedoch ist mir nicht mehr klar, warum.

Reply:
Die Lösung mit Fehlern wurde implementiert. Hat mir jedoch auch gezeigt, dass hier noch Nachbesserungsbedarf besteht. Habe `Search` noch um einen Fehler bei leerem Haystack und bei `len(haytack) < len(needle)` erweitert